respy/state_space.py

```diff
@@ -488,21 +505,28 @@ def _create_state_space_indexer(df, optim_paras):
         shape = (
             tuple(np.minimum(max_initial_experience + period, max_experience) + 1)
             + (n_exp_choices + n_nonexp_choices,) * optim_paras["n_lagged_choices"]
+            + tuple(len(x) for x in optim_paras["observables"].values())
             + (optim_paras["n_types"],)
         )
         sub_indexer = np.full(shape, -1, dtype=np.int32)
 
         sub_df = df.loc[df.period.eq(period)]
         n_states = sub_df.shape[0]
 
-        indices = (
-            tuple(sub_df[f"exp_{i}"] for i in optim_paras["choices_w_exp"])
-            + tuple(
-                sub_df[f"lagged_choice_{i}"].replace(choice_to_code)
-                for i in range(1, optim_paras["n_lagged_choices"] + 1)
-            )
-            + (sub_df.type,)
+        indices = tuple(
+            sub_df[f"exp_{i}"] for i in optim_paras["choices_w_exp"]
+        ) + tuple(
+            sub_df[f"lagged_choice_{i}"].replace(choice_to_code)
+            for i in range(1, optim_paras["n_lagged_choices"] + 1)
         )
+        if "observables" in optim_paras:
```

Comment:
The if statement is pointless. We always insert the key, but there might be nothing under it. The tuple comprehension takes care of the case where not observables are used.